.buildkite/pipeline.yml

```diff
@@ -1,13 +1,13 @@
-publish-android-artifacts-docker-container: &publish-android-artifacts-docker-container
-  plugins:
-    - docker#v3.8.0:
-        image: "public.ecr.aws/automattic/android-build-image:v1.0.0"
-        propagate-environment: true
-        environment:
-          # DO NOT MANUALLY SET THESE VALUES!
-          # They are passed from the Buildkite agent to the Docker container
-          - "AWS_ACCESS_KEY"
-          - "AWS_SECRET_KEY"
+x-common-params:
+  &publish-android-artifacts-docker-container
+  docker#v3.8.0:
+    image: "public.ecr.aws/automattic/android-build-image:v1.0.0"
+    propagate-environment: true
+    environment:
+      # DO NOT MANUALLY SET THESE VALUES!
+      # They are passed from the Buildkite agent to the Docker container
+      - "AWS_ACCESS_KEY"
+      - "AWS_SECRET_KEY"
```

Comment:
Maybe it's premature "optimization" from my part, but in case we need to add more anchors in the future, I feel like it could make more sense to make the `common-params` dummy dict contain an array (one item per anchor we want to define), rather than directly the sole anchor we currently have. This way if we need to define more anchors as we make the YAML evolve, we can just add another item to the array.

```suggestion
  - &publish-android-artifacts-docker-container
    docker#v3.8.0:
      image: "public.ecr.aws/automattic/android-build-image:v1.0.0"
      propagate-environment: true
      environment:
        # DO NOT MANUALLY SET THESE VALUES!
        # They are passed from the Buildkite agent to the Docker container
        - "AWS_ACCESS_KEY"
        - "AWS_SECRET_KEY"
```

Another option is to make it a dictionary, with the keys being the same keys that we'll use at call site (e.g. `plugins:` for this one), as a kind of reminder of which key this anchor is supposed to be used to in the rest of the YAML.

```suggestion
  plugins: &publish-android-artifacts-docker-container
    docker#v3.8.0:
      image: "public.ecr.aws/automattic/android-build-image:v1.0.0"
      propagate-environment: true
      environment:
        # DO NOT MANUALLY SET THESE VALUES!
        # They are passed from the Buildkite agent to the Docker container
        - "AWS_ACCESS_KEY"
        - "AWS_SECRET_KEY"
```

Though while it helps remind us where this anchor will then be used, that might also bring back the risk of confusion for people not used to anchors in thinking the key name before the anchor would be meaningful while it's not. Maybe a middle ground would be to pick an arbitrary key name prefixed with `x-` (aka: `x-plugins`) or suffixed with `key` (aka: `plugins-key:`), or use `common-plugins` or something like this?

All this is really just a brain dump and total nitpick, just opening the can of worms for the higher discussion, but is not a blocker at all for this PR.

Reply:
@AliSoftware I think you might be overthinking this as I am having a very hard time following your thought process. The current iteration is exactly what you previously suggested, so I think I'll just keep it as is because it does exactly what I want for the current moment.

I am not saying let's never change it, but when it's laid out like this - especially with the back & forth we already had - it's becoming less constructive. My suggestion is, if you prefer a certain way and you believe there is a good reason for it:

**a.** Open a new PR so the description explains why we made that choice
**b.** Publish an RFC in the internal p2 so we can make a decision and internally document it in the `Automattic/buildkite-ci` so we all follow the same practice.
**c.** Open a PR in `Automattic/buildkite-ci` with YAML best practices, so we can discuss it there and then start applying it everywhere.

Regardless of the option(s) you go with, I'd suggest limiting the alternatives because if we have a lot of them, not only is it confusing and hard to track the thought process, but also it means that there is no clear winner between them.

I hope that makes sense to you!